chia/seeder/dns_server.py

```diff
@@ -1,294 +1,365 @@
             try:
-                # TODO: double check this. It shouldn't take this long to connect.
-                crawl_db = await aiosqlite.connect(self.db_path, timeout=600)
-                cursor = await crawl_db.execute(
-                    "SELECT * from good_peers",
-                )
-                new_reliable_peers = []
-                rows = await cursor.fetchall()
-                await cursor.close()
-                await crawl_db.close()
-                for row in rows:
-                    new_reliable_peers.append(row[0])
-                if len(new_reliable_peers) > 0:
-                    random.shuffle(new_reliable_peers)
-                async with self.lock:
-                    self.reliable_peers_v4 = []
-                    self.reliable_peers_v6 = []
-                    for peer in new_reliable_peers:
-                        ipv4 = True
+                new_reliable_peers = await self.crawl_store.get_good_peers()
+            except Exception as e:
+                log.error(f"Error loading reliable peers from database: {e}. TracebacK: {traceback.format_exc()}.")
+            if len(new_reliable_peers) == 0:
+                await asyncio.sleep(60)  # sleep for 1 minute, because the crawler has not yet started
+                continue
+            async with self.lock:
+                self.reliable_peers_v4 = []
+                self.reliable_peers_v6 = []
+                self.pointer_v4 = 0
+                self.pointer_v6 = 0
+                for peer in new_reliable_peers:
+                    ipv4_peer = None
+                    try:
+                        ipv4_peer = IPv4Address(peer)
+                    except ValueError:
+                        pass
+                    if ipv4_peer is not None:
+                        self.reliable_peers_v4.append(ipv4_peer)
+                    else:
                         try:
-                            _ = ipaddress.IPv4Address(peer)
+                            ipv6_peer = IPv6Address(peer)
                         except ValueError:
-                            ipv4 = False
-                        if ipv4:
-                            self.reliable_peers_v4.append(peer)
-                        else:
-                            try:
-                                _ = ipaddress.IPv6Address(peer)
-                            except ValueError:
-                                continue
-                            self.reliable_peers_v6.append(peer)
-                    self.pointer_v4 = 0
-                    self.pointer_v6 = 0
+                            log.error(f"Invalid peer: {peer}")
+                            continue
+                        self.reliable_peers_v6.append(ipv6_peer)
                 log.error(
                     f"Number of reliable peers discovered in dns server:"
                     f" IPv4 count - {len(self.reliable_peers_v4)}"
                     f" IPv6 count - {len(self.reliable_peers_v6)}"
                 )
```

Comment:
If you're moving everything else to INFO you probably want to move these too.

Reply:
nice catch, thanks.